src/haptic_feedback.c

```diff
@@ -0,0 +1,127 @@
+// Copyright 2024 Lukas Hrazky
+//
+// This file is part of the Refloat VESC package.
+//
+// Refloat VESC package is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your
+// option) any later version.
+//
+// Refloat VESC package is distributed in the hope that it will be useful, but
+// WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for
+// more details.
+//
+// You should have received a copy of the GNU General Public License along with
+// this program. If not, see <http://www.gnu.org/licenses/>.
+
+#include "haptic_feedback.h"
+
+#include "vesc_c_if.h"
+
+#include <math.h>
+
+void haptic_feedback_init(HapticFeedback *hf) {
+    hf->type_playing = HAPTIC_FEEDBACK_NONE;
+    hf->start_time = 0.0f;
+    hf->is_playing = false;
+}
+
+void haptic_feedback_configure(HapticFeedback *hf, const CfgHapticFeedback *cfg) {
+    hf->cfg = cfg;
+}
+
+static HapticFeedbackType state_to_haptic_type(const State *state) {
+    if (state->state != STATE_RUNNING) {
```

Comment:
You're gonna want to check if state->mode == MODE_FLYWHEEL so duty cycle haptics don't happen in flywheel mode.

Reply:
Duty SAT is not being set during Flywheel (and I tested it, it doesn't buzz for me). Not sure about the error ones right now. If they were:
- you'd supposedly want to hear them
- they'll most likely make the board fall over as they change the setpoint

Reply:
Sounds good. Only reason I mention is because it was in the original float package.